### PR TITLE
Fix golint failures of e2e/framework/util.go - part3

### PR DESCRIPTION
**Pull request by atoato88**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This is a part of a series for fixing golint failures for util.go.
- fixes golint failures from line 2354 to line 3685 at original util.go

This fixes golint failures of the following file:
- test/e2e/framework/util.go

**Special notes for your reviewer**:

The others of this series are following:

- part1: #76348
- part2: #76488

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Ref: #68026

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -2429,6 +2432,7 @@ func dumpAllNodeInfo(c clientset.Interface) {
 	DumpNodeDebugInfo(c, names, Logf)
 }
 
+// DumpNodeDebugInfo dumps debug information of given nodes.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]: Thanks @atoato88 
added a couple of minor comments.
```suggestion
// DumpNodeDebugInfo dumps debug information of the given nodes.
```

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -2640,6 +2644,7 @@ func WaitForAllNodesSchedulable(c clientset.Interface, timeout time.Duration) er
 	})
 }
 
+// GetPodSecretUpdateTimeout reuturns a duration for syncing a data.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// GetPodSecretUpdateTimeout reuturns the timeout duration for updating pod secret.
```

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -2676,10 +2681,12 @@ func getNodeTTLAnnotationValue(c clientset.Interface) (time.Duration, error) {
 	return time.Duration(intValue) * time.Second, nil
 }
 
+// AddOrUpdateLabelOnNode adds the given label key and value to the given node or updates value.
 func AddOrUpdateLabelOnNode(c clientset.Interface, nodeName string, labelKey, labelValue string) {
 	ExpectNoError(testutils.AddLabelsToNode(c, nodeName, map[string]string{labelKey: labelValue}))
 }
 
+// AddOrUpdateLabelOnNodeAndReturnOldValue adds the given label key and value to the given node or updates value and returns an old label value.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// AddOrUpdateLabelOnNodeAndReturnOldValue adds the given label key and value to the given node or updates value and returns the old label value.
```

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -2689,18 +2696,21 @@ func AddOrUpdateLabelOnNodeAndReturnOldValue(c clientset.Interface, nodeName str
 	return oldValue
 }
 
+// ExpectNodeHasLabel expects that a value for the given key of the given node equals to the given label value.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// ExpectNodeHasLabel expects that the given node has the given label pair.
```

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -2805,6 +2817,7 @@ func RemoveAvoidPodsOffNode(c clientset.Interface, nodeName string) {
 	ExpectNoError(err)
 }
 
+// ScaleResource scales pods to the given size.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// ScaleResource scales resource to the given size.
```

File: test/e2e/framework/util.go [hunk 6]
```diff
@@ -3374,6 +3391,7 @@ func CreatePodOrFail(c clientset.Interface, ns, name string, labels map[string]s
 	ExpectNoError(err, "failed to create pod %s in namespace %s", name, ns)
 }
 
+// DeletePodOrFail deletes a pod of the specified namespace and name.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// DeletePodOrFail deletes the pod of the specified namespace and name.
```

File: test/e2e/framework/util.go (hunk 1 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 2 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 3 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 4 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 5 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 6 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done. 😄

File: test/e2e/framework/util.go [hunk 7]
```diff
@@ -3455,41 +3473,38 @@ func isNodeConditionSetAsExpected(node *v1.Node, conditionType v1.NodeConditionT
...
-						if !silent {
-							Logf(msg)
-						}
-						return false
 					}
-				} else {
-					// TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default
-					if cond.Status != v1.ConditionTrue {
-						return true
+					msg := ""
```
Comment (timothysc) [inline test/e2e/framework/util.go]: So your PR to this point was pure linting, but some of this looks like style choice, and I'd argue the large else{} block is easier to read.  I get why you eliminated due to the returns, but when reading a mountain of code fast else{}'s help.

**Review by timothysc — COMMENTED** [review]
I'd restructure differently for better readability

/approve 

hold on lgtm till change.

File: test/e2e/framework/util.go (hunk 7 above)
Comment (oomichi) [reply test/e2e/framework/util.go]: I can see @timothysc's point, but that could depend on developers' background and thinking.
That means sometimes it is difficult to make concrete consensus and that is reason why we have golint check to make it simple and easy.
Ideally it is great to have some skipping flag for golang like python's `#noqa` for enabling linting check with partial skip.

Comment (atoato88) [reply test/e2e/framework/util.go]: Thank you for reviewing.
I returned this part to original.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Pull request merged**